cpp/src/reductions/scan/ewm.cu

```diff
@@ -0,0 +1,382 @@
+    the normalization factors and divide the results into them at the end.
+    */
+    auto device_view = column_device_view::create(input);
+    auto valid_it    = detail::make_validity_iterator(*device_view);
+
+    auto data = thrust::make_zip_iterator(thrust::make_tuple(
+      input.begin<T>(), thrust::make_counting_iterator<size_type>(0), valid_it, nullcnt.begin()));
+
+    thrust::transform_inclusive_scan(rmm::exec_policy(stream),
+                                     data,
+                                     data + input.size(),
+                                     pairs.begin(),
+                                     ewma_noadjust_functor<T, true>{beta},
+                                     recurrence_functor<T>());
+  }
+
+  // copy the second elements to the output for now
+  thrust::transform(rmm::exec_policy(stream),
+                    pairs.begin(),
+                    pairs.end(),
+                    output.begin(),
+                    [] __device__(pair_type<T> pair) -> T { return pair.second; });
+  return output;
+}
+
+template <typename T>
+std::unique_ptr<column> ewma(scan_aggregation const& agg,
+                             column_view const& input,
+                             rmm::cuda_stream_view stream,
+                             rmm::mr::device_memory_resource* mr)
+{
+  auto ewma_agg = (dynamic_cast<ewma_aggregation const*>(&agg));
+  if (ewma_agg == NULL) { CUDF_FAIL("Expected an EWMA aggregation."); }
+  CUDF_EXPECTS(cudf::is_floating_point(input.type()), "Column must be floating point type");
+
+  cudf::ewm_history const history = ewma_agg->history;
+  T const center_of_mass          = ewma_agg->center_of_mass;
+
+  // center of mass is easier for the user, but the recurrences are
+  // better expressed in terms of the derived parameter `beta`
+  T const beta = center_of_mass / (center_of_mass + 1.0);
+
+  auto result = [&]() {
+    if (history == cudf::ewm_history::INFINITE) {
+      return compute_ewma_adjust(input, beta, stream, mr).release();
+    } else {
+      return compute_ewma_noadjust(input, beta, stream, mr).release();
+    }
+  }();
+  return std::make_unique<column>(
+    cudf::data_type(cudf::type_to_id<T>()), input.size(), std::move(result));
+}
+
+struct ewma_functor {
+  template <typename T>
+  std::enable_if_t<!is_floating_point<T>(), std::unique_ptr<column>> operator()(
+    scan_aggregation const& agg,
+    column_view const& input,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    CUDF_FAIL("Unsupported type for EWMA.");
+  }
+
+  template <typename T>
+  std::enable_if_t<is_floating_point<T>(), std::unique_ptr<column>> operator()(
+    scan_aggregation const& agg,
+    column_view const& input,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    return ewma<T>(agg, input, stream, mr);
+  }
+};
+
+std::unique_ptr<column> exponential_weighted_moving_average(column_view const& input,
```

Comment:
I think the term is usually "exponential_ly_ weighted". https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average

Or use "exponential moving average" if you want to go for EMA instead of EWMA everywhere. Anything consistent is fine with me.

```suggestion
std::unique_ptr<column> exponentially_weighted_moving_average(column_view const& input,
```